perspective_rectification/hough_transform_intersections.py

```diff
@@ -96,7 +96,7 @@ def average_lines_clustered(lines, labels):
         x_intercept = np.median(list(list(zip(*filtered_lines[j]))[0]))
         y_intercept = np.median(list(list(zip(*filtered_lines[j]))[1]))
         # TODO : we can try using median with theta too
-        theta = np.average(list(list(zip(*filtered_lines[h]))[2]))
+        theta = np.average(list(list(zip(*filtered_lines[j]))[2]))
```

Comment:
Why the double list() call? (here and in the surrounding lines)

Reply:
i think it was because the first paring of the list is to extract the zip and the second one to make the average of all the resulting filtered lines

Reply:
But list() converts another iterable (e.g. a tuple) to a list, so `list(list((1,2,3))` still yields only `[1,2,3]`, as the first call to `list()` already converts the tuple to a list and the second call doesn't do anything extra. If we wanted a nested list here, we should use brackets for the outer list: `[list(zip(*filtered_lines[j])]`.

Reply:
fixed